plugins/prometheus/pkg/data/data.go

```diff
@@ -92,6 +91,28 @@ func getController(workloads []controller.Workload, podName, namespace string) (
 	return
 }
 
+func GetNodesMetrics(ctx context.Context, dynamicClient dynamic.Interface, restMapper meta.RESTMapper, api prometheusV1.API) (*NodesMetrics, error) {
+	r := getRange()
+	idleCPU, err := getNodesIdleCPU(ctx, api, r)
+	if err != nil {
+		return nil, err
+	}
+	logrus.Infof("Found %d metrics for idle CPU", len(idleCPU))
+	idleMemory, err := getNodesIdleMemory(ctx, api, r)
+	if err != nil {
+		return nil, err
+	}
+	logrus.Infof("Found %d metrics for idle Memory", len(idleMemory))
+	response := &NodesMetrics{}
+	if len(idleCPU) > 0 {
+		response.IdleCPU = idleCPU[0].Values[0].Value
```

Comment:
is `Values[0]` always there or requires a length check?

Reply:
It should be but better adding a check